Update cross-reference-checker

gasche opened the pull request:
This useful piece of code was not built during our testing, and has gone unbuildable due to two separate changes to the parsetree in #1952 and #1953. (When it doesn't build, building the manual fails.)

The present PR fixes the two build errors, and adds building cross-reference-checker to the `manual-pregen` step of the CI, so that we notice future breakages earlier.

@Armael: I added a code-migration remark to the Changes entry for #1952.

gasche committed:
cross-reference-checker: fix Location.report_error after GPR #1952

gasche committed:
cross-reference-checker: fix attribute structure after GPR #1953

gasche commented:
(cc @Octachron)

Armael commented on Changes:
```diff
@@ -252,6 +252,7 @@ Working version
 ### Internal/compiler-libs changes:
 
 * GPR#1952: refactor the code responsible for displaying errors and warnings
+  `Location.report_error` is removed, use `Location.print_error` instead
```
It's rather `Location.print_report` right?

gasche replied:
Indeed, thanks.

gasche committed:
build cross-reference-checker during manual-pregen

Many build errors in cross-reference-checker have gone unnoticed
because this program was only built to build the manual itself, which
is not part of the CI testing. (The test that it implements needs the
.aux files produces by LaTeX, which we don't really want to produce
during CI runs.)

This commit adds cross-reference-checker to the `tools` target of
manual/Makefile, which is in turn built by the `manual-pregen` target
of the main Makefile, which contains the sanity checks for the manual
that do not require building it, and is part of our CI.

Octachron reviewed (approved):
The fixes themselves look good to me and building the cross-reference-checker before manual-pregen sounds like a good idea.

gasche commented:
Thanks! Merged.
(I'm in the process of rebasing my manual-examples PR so I didn't want to wait too long.)
